src/Compilers/Core/Portable/WellKnownTypes.cs

```diff
@@ -303,6 +303,11 @@ internal enum WellKnownType
         System_InvalidOperationException,
         System_Runtime_CompilerServices_SwitchExpressionException,
 
+        System_Diagnostics_CodeAnalysis_DisallowNullAttribute,
```

Comment:
>System_Diagnostics_CodeAnalysis_DisallowNullAttribute, [](start = 8, length = 54)

Addition of these types look suspicious, I thought compiler never synthesizes these attributes.

Reply:
We need those types to be able to declare well-known members (the constructors for those 4 attributes). We synthesize those on accessors when an attribute is present in source on the property.
See `SynthesizedAccessorValueParameterSymbol.cs` which does `AddSynthesizedAttribute(ref attributes, compilation.TrySynthesizeAttribute(WellKnownMember.System_Diagnostics_CodeAnalysis_DisallowNullAttribute__ctor));` (there is similar code for the other 3 attributes)


---
In reply to: [295417296](https://github.com/dotnet/roslyn/pull/36444#discussion_r295417296) [](ancestors = 295417296)

Reply:
> We synthesize those on accessors when an attribute is present in source on the property.

We should use attributes from the property instead of inventing new ones.

---
In reply to: [295491504](https://github.com/dotnet/roslyn/pull/36444#discussion_r295491504) [](ancestors = 295491504,295417296)